### PR TITLE
Add `torch.unravel_index`

### LINKED ISSUE #35674: Function request: unravel_index
## 🚀 Feature

This function would mirror `np.unravel_index`; a basic Torch implementation and todo to formally expose such an implementation were previously added [here](https://github.com/pytorch/pytorch/blob/c73e97033a3aef97a5685588ea014d54a5cc11cc/torch/testing/__init__.py#L41-L49).
A few more suggestions for implementation may be found at this [SO thread](https://stackoverflow.com/questions/53212507/how-to-efficiently-retrieve-the-indices-of-maximum-values-in-a-torch-tensor).

## Motivation

As per the SO thread, functions like `argmax` now return a flat index, while sometimes one may want an unravel (/ multi-dimensional) index.

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @mruberry @rgommers @heitorschueroff

### PR DESCRIPTION
**Pull request by kurtamohler**
Fixes #35674

cc @mruberry @rgommers @albanD

### CODE REVIEW

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -1624,6 +1625,49 @@ Tensor alias_with_sizes_and_strides(
   return self_;
 }
 
+Tensor _unravel_index(const Tensor& self, c10::IntArrayRef shape) {
+  TORCH_CHECK(
+      c10::isIntegralType(self.scalar_type(), /*includeBool=*/false),
+      "expected integer indices, but got", self.scalar_type());
+  Tensor shape_tensor = at::empty(
```
Comment (kurtamohler) [inline aten/src/ATen/native/TensorShape.cpp:1632]: TODO: replace with `at::from_blob`

**Review by albanD — COMMENTED** [review]
Why is this not the simple python implementation from the issue?

**Comment by kurtamohler** [comment]
> Why is this not the simple python implementation from the issue?

I thought we would want it in aten. If not, I can just do a python implementation

**Comment by albanD** [comment]
I think we should keep it in python unless we have a good reason to move it to aten.

**Comment by kurtamohler** [comment]
Closing this in favor of #110580

**Pull request closed without merging**